### PR TITLE
Fix for non-interactive docker output and docker attach

### PR DESCRIPTION
**Pull request by amshinde**
This is a workaround for using stdio as console device for qemu.
This solution bypasses the shim pipes and writes to docker named
pipes. We should resort to using stdio as console once that is resolved.

### CODE REVIEW

**Comment by grahamwhaley** [comment]
Hi. Is this related to #17 and/or #34 ?  If so, let's consider referencing them in the commit message (whilst I contemplate if that is the right thing to do for github commits...)

**Comment by amshinde** [comment]
grahamwhaley This is related #17

**Comment by sameo** [comment]
@amshinde @grahamwhaley If it fixes #17, it should appear in the commit message:

```
Fixes #17
```

And github will close the issue with a reference to the commit automatically.

**Comment by amshinde** [comment]
sameo, this solution works. But I'd like to keep the #17 open until the stdio problem with qemu is figured out

**Comment by devimc** [comment]
:+1:

File: src/hypervisor.c [hunk 1]
```diff
@@ -124,23 +126,76 @@ cc_oci_expand_cmdline (struct cc_oci_config *config,
...
+				g_critical("Failed to create symlink for input pipe: %s",
+					   strerror (errno));
+				goto out;
+			}
+
+			console_device = g_strdup_printf ("pipe,id=charconsole0,path=%s", config->console);
+
+		} else {
+
+			/* In case the runtime is called standalone without console
```
Comment (grahamwhaley) [inline src/hypervisor.c]: nit pick - this could be a single line comment I think

File: src/hypervisor.c [hunk 2]
```diff
@@ -124,23 +126,76 @@ cc_oci_expand_cmdline (struct cc_oci_config *config,
...
+			*/
+
+			/* No console specified, so make the hypervisor create
```
Comment (grahamwhaley) [inline src/hypervisor.c]: nit pick - this comment and the one following it are in different 'styles' - with the alignment/indentation of the '*'s differing. We should stick to one style (normally the style already in use in the file/project).

**Comment by grahamwhaley** [comment]
Hi @amshinde . I just had a couple of comment style nitpicks - if you can fix them up trivially then I'll merge this.
 thanks!

**Comment by amshinde** [comment]
grahamwhaley, I pushed the changes you suggested.

**Pull request merged**